face/infra/bio-sdk-resolver/src/main/java/com/simprints/face/infra/biosdkresolver/ResolveFaceBioSdkUseCase.kt

```diff
@@ -1,28 +1,16 @@
 package com.simprints.face.infra.biosdkresolver
 
-import com.simprints.infra.config.store.ConfigRepository
 import com.simprints.infra.config.store.models.FaceConfiguration
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-class ResolveFaceBioSdkUseCase @Inject constructor(
-    private val configRepository: ConfigRepository,
-    private val rocV1BioSdk: RocV1BioSdk,
-    private val rocV3BioSdk: RocV3BioSdk,
-    private val simFaceBioSdk: SimFaceBioSdk,
+class ResolveFaceBioSdkUseCase @Inject internal constructor(
+    private val resolveRankOneVersionUseCase: ResolveRankOneVersionUseCase,
+    private val resolveSimFaceVersionUseCase: ResolveSimFaceVersionUseCase,
 ) {
     suspend operator fun invoke(bioSdk: FaceConfiguration.BioSdk): FaceBioSDK = when (bioSdk) {
-        FaceConfiguration.BioSdk.SIM_FACE -> simFaceBioSdk
-        FaceConfiguration.BioSdk.RANK_ONE -> {
-            val version = configRepository
-                .getProjectConfiguration()
-                .face
-                ?.rankOne
-                ?.version
-                ?.takeIf { it.isNotBlank() } // Ensures version is not null or empty
-            requireNotNull(version) { "FaceBioSDK version is null or empty" }
-            if (version == rocV3BioSdk.version) rocV3BioSdk else rocV1BioSdk
-        }
+        FaceConfiguration.BioSdk.SIM_FACE -> resolveSimFaceVersionUseCase()
+        FaceConfiguration.BioSdk.RANK_ONE -> resolveRankOneVersionUseCase()
```

Comment:
It's a matter of preference but do we really need to extract something as simple as the version resolving? It seems the overhead is higher than the code itself.

Reply:
It is not necessary, but it has the benefit of isolating each SDK's logic. 

Honestly, I was just trying a lot of things to minimise the amount of classes that we would need to duplicate per build type and this is a by-product of the experimentation. We can simplify it when the need for build types is gone.